kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/SharedInformerFactory.java

```diff
@@ -84,21 +104,22 @@ public SharedInformerFactory(OkHttpClient okHttpClient, Config configuration) {
    * @param configuration configuration for client
    */
   public SharedInformerFactory(ExecutorService threadPool, OkHttpClient okHttpClient, Config configuration) {
-    super(new OperationContext().withOkhttpClient(okHttpClient).withConfig(configuration));
+    this.baseOperation = new SharedInformerFactoryBaseOperation(new OperationContext().withOkhttpClient(okHttpClient).withConfig(configuration));
     initOperationContext(configuration);
     this.informerExecutor = threadPool;
-    this.namespace = null;
+    this.baseOperation.setNamespace(null);
   }
 
   /**
    * Configure Namespace for {@link SharedInformerFactory}
    *
    * @param namespace namespace to configure
    * @return {@link SharedInformerFactory} with namespace configured
+   * @deprecated use {@link Informable} instead
    */
-  @Override
+  @Deprecated
   public SharedInformerFactory inNamespace(String namespace) {
```

Comment:
Umm, why are we deprecating this? I think using namespaced informers with SharedInformerFactory can be a very common use case. Even client-go's informer factory provides Namespaced informers via [`NewFilteredSharedInformerFactory`/`NewSharedInformerFactoryWithOptions`](https://github.com/kubernetes/client-go/blob/master/informers/factory.go#L109)

Reply:
It behaves differently than the withName and inNamespace dsl methods - it modifies the current context rather than creating a new one.  I also find this behavior confusing wrt an informer factory - if you can set this post construction, then some informers can end watching all namesapces, while others would be in a specific namespace.  Also a inAnyNamespace is not implemented.  At least to me it seems like these methods are problematic.

Reply:
I agree mutating context is not good. We should think about providing similar alternatives to the above-mentioned client-go methods in the future.

Reply:
Direct alternatives would be setName and setNamespace - but I still wonder if it may not be better to offer either:

Informable.informers() - where the factory is off of the current context.

or 

KubernetesClient.informers(InformerOptions) - with options similar to the go client.

Neither of those require additional methods on the factory.